### PR TITLE
Pending pacts

### PR DESCRIPTION
**Pull request by jbwheatley**

### CODE REVIEW

**Comment by jbwheatley** [comment]
#160

File: scalapact-shared/src/main/scala/com/itv/scalapact/shared/PactsForVerificationResponse.scala [hunk 1]
```diff
@@ -34,14 +35,40 @@ import com.itv.scalapact.shared.Pact.Links
...
+
+  final case class PendingStateVerificationProperties(pending: Boolean, notices: List[PendingStateNotice]) extends VerificationPropertiesAux[PendingStateNotice]
+
+}
+
+sealed trait Notice {
+  def text: String
+}
+
+object Notice {
```
Comment (bethesque) [inline scalapact-shared/src/main/scala/com/itv/scalapact/shared/PactsForVerificationResponse.scala:69]: The logic should really only care about "before verification" and "after verification" notices. I want to be able to put anything in here, and have it display automatically without any client changes. `PendingStateNotice` seems to be putting business logic into the code that I don't think it should know about.

**Review by bethesque — COMMENTED** [review]
Awesome work! Thanks so much.

I can't read scala well enough to understand what logic has been used for calculating the exit status of the verification task. Essentially, the overall verification task should now only fail if a non-pending pact fails verification (there are likely to be a mix of pending and non pending, and non-pending failures still need to fail the overall build). Is this what has been implemented?

Just checking also that the results of the pending test are reported back to the broker accurately - if a pending pact fails, it's still a failure in the broker, not a success. The exit status massaging only happens on the provider side.

**Comment by jbwheatley** [comment]
Hey @bethesque, thanks for taking a look! Glad to have an experienced pair of eyes on it. 

>  the overall verification task should now only fail if a non-pending pact fails verification

Yep, this is what has been implemented. Also failures of pending pacts are still reported  to the pact broker in the same way as non-pending failures have been (if publishing results is enabled).

> `PendingStateNotice` seems to be putting business logic into the code that I don't think it should know about.

You're totally right, this naming is too specific. I'll change it 😄

**Pull request merged**